compiler/modulegraphs.nim

```diff
@@ -343,31 +364,44 @@ proc hash*(u: SigHash): Hash =
 
 proc hash*(x: FileIndex): Hash {.borrow.}
 
+template getC(): untyped =
+  when c is PContext: c
+  else: c.c
+
+template onDefAux(info: TLineInfo; s0: PSym, c0: untyped, isFwd: bool) =
+  if s0.kind in ExportableSymKinds:
+    let c = c0 # in case c0 is an expression
+    var top = true
+    case s0.kind
+    of skProc, skMacro:
+      # unfortunately, can't use `c.isTopLevel` because the scope isn't closed yet
+      top = c.currentScope.depthLevel <= 3
+    else: top = c.currentScope.depthLevel <= 2
+    if top and c.module != nil:
+      strTableAdd(semtabAll(c.graph, c.module), s0)
+      if c.config.symbolFiles != disabledSf:
+        addHidden(c.encoder, c.packedRepr, s0)
```

Comment:
Same ... too ugly.

Reply:
I've now improved this:
* template => proc
* improve control flow, simplify
* don't call this for onDefResolveForward (and remove isFwd), it wasn't needed in this case

Reply:
i'm now using addInterfaceDeclAux instead